### PR TITLE
[CSL-1317] Integrate pure db into tests

### PR DESCRIPTION
**Pull request by volhovM**
Subj. Now pure db is used in block test mode. 

Another way would be to modify `InitContextMode`, but i tried it and it didn't work out.

### CODE REVIEW

File: src/Pos/DB/Block.hs [hunk 1]
```diff
@@ -266,6 +269,24 @@ dbPutBlundPureDefault (blk,undo) = do
...
+    -> m (Maybe (SscBlock ssc))
+dbGetBlockSscPureDefault = fmap (toSscBlock <$>) . dbGetBlockPureDefault
+
+dbGetUndoSscPureDefault ::
+       forall ssc ctx m. (MonadPureDB ctx m, SscHelpersClass ssc)
+    => HeaderHash
+    -> m (Maybe ())
+dbGetUndoSscPureDefault = fmap (const () <$>) . dbGetUndoPureDefault @ssc
+
+dbGetHeaderSscPureDefault ::
```
Comment (gromakovsky) [inline src/Pos/DB/Block.hs]: Btw, the functions in this module shouldn't have `Default` suffix, but it's minor.

File: test/Test/Pos/Block/Logic/Mode.hs [hunk 2]
```diff
@@ -152,11 +146,32 @@ instance Arbitrary TestParams where
...
+-- Init mode with instances
+----------------------------------------------------------------------------
+
+-- The fields are lazy on purpose: this allows using them with
+-- futures.
+data TestInitModeContext ssc = TestInitModeContext
+    { imcDBPureVar          :: DBPureVar
+    , imcGenesisUtxo        :: GenesisUtxo
+    , imcSlottingVar        :: (Timestamp, TVar SlottingData)
+    , imcSlottingContextSum :: SlottingContextSum
```
Comment (gromakovsky) [inline test/Test/Pos/Block/Logic/Mode.hs]: It's not needed here.

File: test/Test/Pos/Block/Logic/Mode.hs [hunk 3]
```diff
@@ -180,22 +195,21 @@ makeLensesWith postfixLFields ''BlockTestContext
...
-            let initCtx =
-                    InitModeContext
-                        nodeDBs
-                        tpGenUtxo
-                        futureSlottingVar
-                        SCSimple
-                        futureLrcCtx
-            runInitMode @SscGodTossing initCtx $
-                bracketBlockTestContextDo nodeDBs putSlottingVar putLrcCtx
+bracketBlockTestContext testParams@TestParams {..} callback = do
```
Comment (gromakovsky) [inline test/Test/Pos/Block/Logic/Mode.hs]: It's not bracket anymore and probably will never be bracket.

File: src/Pos/DB/Block.hs (hunk 1 above)
Comment (volhovM) [inline src/Pos/DB/Block.hs]: Why not? There are lots of `aoeuDefault`, I've added `aoeuPureDefault`s also.

Comment (gromakovsky) [inline src/Pos/DB/Block.hs]: `Default` suffix is used when there is only one implementation. But here we have `Pure` and `Real`/`Rocks` implementations. So `Default` suffix is redundant.

Comment (volhovM) [inline src/Pos/DB/Block.hs]: I thought that `Default` suffix means "put this method in the instance implementation" which makes a lot of sense for me.

**Pull request merged**